build-logic-commons/publishing/src/main/kotlin/gradlebuild.kotlin-dsl-plugin-bundle.gradle.kts

```diff
@@ -123,7 +123,11 @@ gradlePlugin {
         }
 
         writeFuturePluginVersions {
-            property(plugin.id, version)
+            /**
+             * TODO: Remove for Gradle 9.0, used as a workaround for Provider API
```

Comment:
❓ 

So this is meant as a strategy to be source compatible with both Gradle 8 and 9 (when it becomes moot and just a warning for shadowing `Provider.get())).

Reply:
Yes, this is a workaround so it works with Gradle 8 and Gradle 9 until we use Provider API incompatible wrapper. 
Since we use Gradle 8 wrapper `plugin.id` is a String. But in Gradleception tests it's a `Property<String>`. 

So this:
```
fun String.get() = this
```
and calling `plugin.id.get()` always is a trick to be source compatible. I didn't think of a different simple solution, but if you have one let me know